src/usr/share/gpgmailer/gpgkeyverifier.py

```diff
@@ -93,68 +94,55 @@ def get_expiration_warning_message(self, loop_current_time):
-            # TODO: Eventually, handle multiple keys for one address.
-            if recipient['email'] in self.all_recipient_emails:
-                raise RecipientEmailCollision('Email %s is already configured.' % recipient['email'])
-
-            email_dict = { 'fingerprint': recipient['fingerprint'],
+            key_dict = {
+                'fingerprint': fp_to_key_dict[fingerprint]['fingerprint'],
+                'email_list': fp_to_key_dict[fingerprint]['emails'],
                 'expired_email_sent': False,
-                'expiring_soon_email_sent': False,
-                'is_sender': False,
-                'is_recipient': True }
-
-            recipient_fingerprints.append(recipient['fingerprint'])
-            self.email_dicts[recipient['email']] = email_dict
-            self.all_recipient_emails.append(recipient['email'])
+                'expiring_soon_email_sent': False
+            }
 
-        # The sender might also be a receipient.
-        self.sender_email = config['sender']['email']
-        if self.sender_email in self.email_dicts.keys():
-            # The sender is also a recipient.
-            self.email_dicts[self.sender_email]['is_sender'] = True
+            # Mark the gpgmailer configured sender key.
+            if fp_to_key_dict[fingerprint]['fingerprint'] == config['sender']['fingerprint']:
+                self.sender_key = key_dict
 
-            if config['sender']['fingerprint'] not in recipient_fingerprints:
-                raise RecipientEmailCollision('Email %s is already configured with a different key.' % recipient['email'])
+            self.recipient_keys[fp_to_key_dict[fingerprint]['fingerprint']] = key_dict
 
-        else:
-            # The sender is NOT a recipient.
-            email_dict = { 'fingerprint': config['sender']['fingerprint'],
-                'expired_email_sent': False,
-                'expiring_soon_email_sent': False,
-                'is_sender': True,
-                'is_recipient': False}
 
-            self.email_dicts[self.sender_email] = email_dict
-
-
-    # Calculates which recipients keys have expired and which have not. Builds a list of currently
-    #   valid recipients and constructs an expiration warning messages for keys that have expired
+    # Calculates which keys in the keyring have expired and which have not. Builds a list of currently
+    #   valid recipient keys and constructs expiration warning messages for keys that have expired
     #   or will be expiring soon.
     #
     # loop_current_time: The Unix time associated with the main program loop from which all
     #   GPG key expiration checks are based.
     def _calculate_recipient_info(self, loop_current_time):
         self.logger.trace('Recalculating the list of keys that are about to expire.')
-        all_expiration_warning_messages = []
         expired_messages = []
         expiring_soon_messages = []
         valid_recipient_emails = []
```

Comment:
I'm thinking valid_recipient_emails should now be the Python equivalent to a java.util.Set.